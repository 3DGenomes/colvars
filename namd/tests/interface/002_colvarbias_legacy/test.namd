# -*- tcl -*-

# Note: see also test.restart.namd, test.legacy.namd, test.restart.legacy.namd

source ../Common/common.namd
source ../Common/measure_net_force_torque.tcl
stepspercycle 5

colvars                 on
colvarsConfig           test.in

if { [file exists test.prerun.tcl] > 0 } {
    source test.prerun.tcl
}

run [expr 5]

print [join [list "Energy difference (centers) = " [colvarbias energydiff "h" "centers 0.0"]]]

print [join [list "Energy difference (forceK) = " [colvarbias energydiff "h" "forceConstant 1.0"]]]

run [expr 10]

colvarbias changeconfig "h" "centers 2.0"

run [expr 5]

if { [file exists test.postprocess.tcl] > 0 } {
    source test.postprocess.tcl
}

source ../Common/delete_tmp_files.tcl
