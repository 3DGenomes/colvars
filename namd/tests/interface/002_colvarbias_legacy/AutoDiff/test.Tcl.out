TCL: Running for 5 steps
TCL: TOTAL FORCE: ( 4.336808689942018e-19 5.421010862427522e-19 -2.6020852139652106e-18 )
TCL: TOTAL TORQUE: ( 1.734723475976807e-18 -1.0408340855860843e-17 4.336808689942018e-18 )
TCL: TOTAL FORCE: ( 2.168404344971009e-19 -1.0842021724855044e-19 4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( -1.0842021724855044e-18 2.0816681711721685e-17 -4.336808689942018e-18 )
TCL: TOTAL FORCE: ( -2.168404344971009e-19 -1.0842021724855044e-19 1.3010426069826053e-18 )
TCL: TOTAL TORQUE: ( -1.3010426069826053e-18 -6.938893903907228e-18 2.6020852139652106e-18 )
TCL: TOTAL FORCE: ( -6.505213034913027e-19 -1.0842021724855044e-19 -4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( -6.505213034913027e-19 1.3877787807814457e-17 -3.469446951953614e-18 )
TCL: TOTAL FORCE: ( 2.168404344971009e-19 0.0 -1.3010426069826053e-18 )
TCL: TOTAL TORQUE: ( 2.168404344971009e-19 1.0408340855860843e-17 -8.673617379884035e-19 )
TCL: Energy difference (centers) =  0.0012624243048076009
TCL: Energy difference (forceK) =  19.275931687793474
TCL: Running for 10 steps
TCL: TOTAL FORCE: ( 2.168404344971009e-19 0.0 4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( 2.168404344971009e-19 3.469446951953614e-18 -8.673617379884035e-19 )
TCL: TOTAL FORCE: ( 2.168404344971009e-19 0.0 4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( 2.168404344971009e-19 3.469446951953614e-18 -8.673617379884035e-19 )
TCL: TOTAL FORCE: ( 2.168404344971009e-19 2.168404344971009e-19 4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( -4.336808689942018e-19 0.0 -1.734723475976807e-18 )
TCL: TOTAL FORCE: ( 0.0 -2.168404344971009e-19 8.673617379884035e-19 )
TCL: TOTAL TORQUE: ( 1.734723475976807e-18 -1.0408340855860843e-17 0.0 )
TCL: TOTAL FORCE: ( 0.0 2.168404344971009e-19 0.0 )
TCL: TOTAL TORQUE: ( -1.3010426069826053e-18 0.0 -8.673617379884035e-18 )
TCL: TOTAL FORCE: ( 0.0 0.0 0.0 )
TCL: TOTAL TORQUE: ( -8.673617379884035e-19 3.469446951953614e-18 3.469446951953614e-18 )
TCL: TOTAL FORCE: ( 0.0 0.0 -4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( 4.336808689942018e-19 1.3877787807814457e-17 -8.673617379884035e-19 )
TCL: TOTAL FORCE: ( 0.0 3.2526065174565133e-19 8.673617379884035e-19 )
TCL: TOTAL TORQUE: ( 0.0 2.0816681711721685e-17 -5.204170427930421e-18 )
TCL: TOTAL FORCE: ( -2.168404344971009e-19 4.336808689942018e-19 8.673617379884035e-19 )
TCL: TOTAL TORQUE: ( -1.3010426069826053e-18 -3.469446951953614e-18 8.673617379884035e-19 )
TCL: TOTAL FORCE: ( -2.168404344971009e-19 -2.168404344971009e-19 -4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( -4.336808689942018e-19 0.0 3.469446951953614e-18 )
TCL: TOTAL FORCE: ( 0.0 -3.2526065174565133e-19 -1.3010426069826053e-18 )
TCL: TOTAL TORQUE: ( 4.336808689942018e-19 -6.938893903907228e-18 0.0 )
TCL: Running for 5 steps
TCL: TOTAL FORCE: ( 0.0 0.0 -4.336808689942018e-19 )
TCL: TOTAL TORQUE: ( 4.336808689942018e-19 3.469446951953614e-18 0.0 )
TCL: TOTAL FORCE: ( 5.421010862427522e-20 0.0 6.505213034913027e-19 )
TCL: TOTAL TORQUE: ( 7.589415207398531e-19 -3.469446951953614e-18 -8.673617379884035e-19 )
TCL: TOTAL FORCE: ( 0.0 -5.421010862427522e-20 0.0 )
TCL: TOTAL TORQUE: ( 0.0 9.540979117872439e-18 -1.5178830414797062e-18 )
TCL: TOTAL FORCE: ( 0.0 2.168404344971009e-19 0.0 )
TCL: TOTAL TORQUE: ( 2.168404344971009e-19 6.938893903907228e-18 -1.951563910473908e-18 )
TCL: TOTAL FORCE: ( -2.168404344971009e-19 -5.421010862427522e-20 0.0 )
TCL: TOTAL TORQUE: ( -5.421010862427522e-19 -5.204170427930421e-18 2.168404344971009e-19 )
TCL: TOTAL FORCE: ( 0.0 -5.421010862427522e-20 -2.168404344971009e-19 )
TCL: TOTAL TORQUE: ( 2.168404344971009e-19 2.6020852139652106e-18 -6.505213034913027e-19 )
